modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbPartitionStorage.java

```diff
@@ -489,21 +495,12 @@ protected DataRow decodeEntry(byte[] key, byte[] value) {
      * and the key's hash (an optimisation).
      */
     private byte[] partitionKey(byte[] key) {
-        byte[] partitionKey = new byte[PARTITION_KEY_PREFIX_SIZE + key.length];
-
-        int pos = 0;
-
-        intToBytes(partId, partitionKey, pos, Integer.BYTES);
-
-        pos += Integer.BYTES;
-
-        intToBytes(Arrays.hashCode(key), partitionKey, pos, Integer.BYTES);
-
-        pos += Integer.BYTES;
-
-        System.arraycopy(key, 0, partitionKey, pos, key.length);
-
-        return partitionKey;
+        return ByteBuffer.allocate(PARTITION_KEY_PREFIX_SIZE + key.length)
+                .order(ByteOrder.BIG_ENDIAN)
+                .putShort((short) partId)
+                .putInt(Arrays.hashCode(key))
```

Comment:
I wonder how I missed this last time. Hash must be pre-calculated in SearchRow as far as I understand. Storage should not control hashing strategy. Default `Arrays.hashCode` isn't good enough.

Reply:
Left a TODO